common/utils.h

```diff
@@ -16,12 +16,16 @@ char *tal_hex(const tal_t *ctx, const tal_t *data);
 /* Allocate and fill a buffer with the data of this hex string. */
 u8 *tal_hexdata(const tal_t *ctx, const void *str, size_t len);
 
-/* Get a temporary context for this function scope (tal_free at end) */
-tal_t *tal_tmpctx_(const tal_t *ctx, const char *file, unsigned int line);
-#define tal_tmpctx(ctx)							\
-	tal_tmpctx_((ctx), __FILE__, __LINE__)
+/* FIXME: Remove in favor of global */
+#define tal_tmpctx(ctx) tal((ctx), char)
 
-/* Return non-NULL if any tmpctx still allocated. */
-const char *tmpctx_any(void);
+/* Global temporary convenience context: freed in io loop core. */
+extern const tal_t *tmpctx;
```

Comment:
In principle, I like this, despite my general horror of static-lifetime objects.

A concern to consider is that the amount of floating, allocated memory that is in-principle "dead" (and which will eventually get cleaned but is not cleaned *now*) increases.  This may be of concern to limited-memory devices such as ARM targets.

I wonder how much of garbage collection we end up implementing.  Currently we effectively make `tmpctx` a singular arena whose contained objects are periodically cleaned, almost like a garbage-collection nursery.  Allocating off `tmpctx` and then `tal_steal`ing them off into a more permanent object parent is almost like garbage-collection promotion.  And so on.